src/tests/ftest/rebuild/container_rf_enforce.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from daos_utils import DaosCommand
+from general_utils import DaosTestError
+from container_redundancy import RbldContRedundancyFactor
+import re
+
+class RbldContRedundancyFactorEnforce(RbldContRedundancyFactor):
+    # pylint: disable=too-many-ancestors
+    """Test Container redundancy factor enforcement with rebuild.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a Rebuild Container RF with ObjClass Write object."""
+        super().__init__(*args, **kwargs)
+        self.mode = None
+        self.daos_cmd = None
+
+    def create_test_container_and_write_obj(self, negative_test=False):
+        """Create a container and write objects
+           for positive testcase, enable the exception with write objects
+           for negative testcase, disable the exception with write objects
+           and expecting failure with RC: -1003.
+
+        Args:
+            negative_test (Bool, optional): container write_objects to handle
+                test Exception and parse for return error code.
+        """
+        der_inval = "RC: -1003"
+        self.log.info(
+            "==>(1)Create pool and container with redundant factor,"
+            " start background IO object write")
+        self.container.create()
+        if not negative_test:
+            self.container.write_objects(
+                self.inputs.rank.value[0], self.inputs.object_class.value)
+        else:
+            try:
+                self.container.write_objects_wo_failon(
```

Comment:
I know how to make dmg not throw exception and get the result output. Not sure if this works with daos but maybe it's worth a try.
```
try:
  self.container.daos.exit_status_exception = False
  self.container.write_objects(...)
  result_stdout = str(self.container.daos.result.stdout)

  if der_inval in str(result_stdout):
    self.log.info("==Negative test, excepting DaosApiError -1003 found")
  else:
    self.fail("#Negative test, container redundancy factor "
                "test failed, return error RC: -1003 not found")
finally:
  self.container.daos.exit_status_exception = True
```
This way, you don't need to create the wo method.

Reply:
Thank for your comment,  I have already tested this.
"self.container.daos.exit_status_exception = False" been overwritten by @fail_on of container.write_objects(...).